links/gql_http_link/lib/src/gql_client.dart

```diff
@@ -0,0 +1,110 @@
+import "package:collection/collection.dart";
+
+class GqlClientRequest {
+  final String method;
+  final Uri uri;
+  final Map<String, dynamic> body;
+  final Map<String, String> headers;
+  final Map<String, dynamic> files;
+  final Map<String, List<String>> fileMapping;
+
+  GqlClientRequest({
+    this.method,
+    this.uri,
+    this.body,
+    this.headers,
+    this.files,
+    this.fileMapping,
+  });
+
+  factory GqlClientRequest.multipart({
+    String method,
+    Uri uri,
+    Map<String, dynamic> body,
+    Map<String, String> headers,
+    Map<String, dynamic> fileMap,
+  }) {
+    final Map<String, List<String>> fileMapping = <String, List<String>>{};
+    final Map<String, dynamic> files = <String, dynamic>{};
+
+    final List<MapEntry<String, dynamic>> fileMapEntries =
+        fileMap.entries.toList(growable: false);
+
+    for (int i = 0; i < fileMapEntries.length; i++) {
+      final MapEntry<String, dynamic> entry = fileMapEntries[i];
+      final String indexString = i.toString();
+      fileMapping.addAll(<String, List<String>>{
+        indexString: <String>[entry.key],
+      });
+      files[indexString] = entry.value;
+    }
+
+    return GqlClientRequest(
+      method: method,
+      uri: uri,
+      body: body,
+      headers: headers,
+      fileMapping: fileMapping,
+      files: files,
+    );
+  }
+
+  bool get isMultipartRequest => files != null && files.isNotEmpty;
+
+  @override
+  bool operator ==(Object o) {
+    if (identical(this, o)) return true;
+    final mapEquals = const DeepCollectionEquality().equals;
+
+    return o is GqlClientRequest &&
+        o.method == method &&
+        o.uri == uri &&
+        mapEquals(o.body, body) &&
+        mapEquals(o.headers, headers) &&
+        mapEquals(o.files, files) &&
+        mapEquals(o.fileMapping, fileMapping);
+  }
+
+  @override
+  int get hashCode =>
```

Comment:
What about inheriting from `gql_exec.Request/Response` and use the same [==/hashCode](https://github.com/gql-dart/gql/blob/7b1dd8a889fff670e6506c020afe2d3fd5866418/links/gql_exec/lib/src/request.dart#L42-L64) pattern we take there?

Reply:
The ==/hashCode pattern there is cleaner.

and how will be handle RequestSerializer and ResponseParser that is included with the link.

I was thinking of it as an apstraction layer above the Request/Response of each http client. inheriting from gql_exec.Request/Response will need the serializer and parser to be also a part of the object I think.